drivers/counter/counter_mcux_rtc.c

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright (c) 2018 blik GmbH
+ * Copyright (c) 2018, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <counter.h>
+#include <fsl_rtc.h>
+
+#define LOG_LEVEL CONFIG_COUNTER_LOG_LEVEL
+#define LOG_MODULE_NAME mcux_rtc
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+struct mcux_rtc_data {
+	const struct counter_alarm_cfg *alarm_cfg;
+};
+
+struct mcux_rtc_config {
+	struct counter_config_info info;
+	RTC_Type *base;
+	void (*irq_config_func)(struct device *dev);
+};
+
+static int mcux_rtc_start(struct device *dev)
+{
+	const struct counter_config_info *info = dev->config->config_info;
+	const struct mcux_rtc_config *config =
+		CONTAINER_OF(info, struct mcux_rtc_config, info);
+
+	RTC_StartTimer(config->base);
+	RTC_EnableInterrupts(config->base,
+			     kRTC_AlarmInterruptEnable |
+			     kRTC_TimeOverflowInterruptEnable |
+			     kRTC_TimeInvalidInterruptEnable);
+
+	return 0;
+}
+
+static int mcux_rtc_stop(struct device *dev)
+{
+	const struct counter_config_info *info = dev->config->config_info;
+	const struct mcux_rtc_config *config =
+		CONTAINER_OF(info, struct mcux_rtc_config, info);
+
+	RTC_DisableInterrupts(config->base,
+			      kRTC_AlarmInterruptEnable |
+			      kRTC_TimeOverflowInterruptEnable |
+			      kRTC_TimeInvalidInterruptEnable);
+	RTC_StopTimer(config->base);
+
+	/* clear out any set alarms */
+	config->base->TAR = 0;
+
+	return 0;
+}
+
+static u32_t mcux_rtc_read(struct device *dev)
+{
+	const struct counter_config_info *info = dev->config->config_info;
+	const struct mcux_rtc_config *config =
+		CONTAINER_OF(info, struct mcux_rtc_config, info);
+
+	u32_t ticks = config->base->TSR;
+
+	/*
+	 * Read TSR seconds twice in case it glitches during an update.
+	 * This can happen when a read occurs at the time the register is
+	 * incrementing.
+	 */
+	if (config->base->TSR == ticks) {
+		return ticks;
+	}
+
+	ticks = config->base->TSR;
+
+	return ticks;
+}
+
+static int mcux_rtc_set_alarm(struct device *dev,
+			      const struct counter_alarm_cfg *alarm_cfg)
+{
+	const struct counter_config_info *info = dev->config->config_info;
+	const struct mcux_rtc_config *config =
+		CONTAINER_OF(info, struct mcux_rtc_config, info);
+	struct mcux_rtc_data *data = dev->driver_data;
+
+	u32_t ticks = alarm_cfg->ticks;
+	u32_t current = mcux_rtc_read(dev);
+
+	LOG_DBG("Current time is %d ticks", current);
+
+	if (data->alarm_cfg != NULL) {
+		LOG_ERR("No more alarms available");
+		return -EINVAL;
+	}
+
+	if (!alarm_cfg->absolute) {
+		ticks += current;
```

Comment:
seems like wrapping is not handled here.

Reply:
No, not yet.